app/main.py

```diff
@@ -4,10 +1,26 @@
+from typing import Dict, Any
 
 
-def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+def format_linter_error(error: dict) -> Dict:
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
+
+
+def format_single_linter_file(file_path: str, errors: list) -> Dict | Any:
```

Comment:
Type Annotation Issue: The return type annotation `Dict | Any` is only supported in Python 3.10 and later. For compatibility with older versions, consider using `Union[Dict, Any]` from the `typing` module.